blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md

```diff
@@ -0,0 +1,384 @@
+---
+title: Implementing the Idempotency-Key specification on Apache APISIX
+authors:
+  - name: Nicolas Fränkel
+    title: Author
+    url: https://github.com/nfrankel
+    image_url: https://avatars.githubusercontent.com/u/752258
+keywords:
+  - APISIX
+  - Idempotency
+  - IETF
+  - specification
+  - plugin
+  - coding
+description: >
+  Last week, I wrote an analysis of the IETF Idempotency-Key specification. The specification aims to avoid duplicated requests. In short, the idea is for the client to send a unique key along with the request:
+  If the server doesn't know the key, it proceeds as usual and then stores the response.
+  If the server knows the key, it short-circuits any further processing and immediately returns the stored response.
+  This post shows how to implement it with Apache APISIX.
+tags: [Community]
+image: https://static.apiseven.com/uploads/2024/04/09/0rfsRevo_stormtrooper-2899993.jpg
+---
+
+<head>
+    <link rel="canonical" href="https://blog.frankel.ch/implement-idempotency-key-apisix/" />
+</head>
+
+>Last week, I wrote an [analysis](https://blog.frankel.ch/fix-duplicate-api-requests/) of the [IETF Idempotency-Key specification](https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-idempotency-key-header-04). The specification aims to avoid duplicated requests. In short, the idea is for the client to send a unique key along with the request:
```

Comment:
Link to the article in the APISIX blog instead? Same for other links that are also available in the APISIX blog.

Reply:
I did it but I'm not sure it's a good idea. Cross domain links improve SEO and my domain is pretty good in that regard

Reply:
Internal links also help as well. But I'm also not sure what would be best.

Reply:
Me neither